### PR TITLE
Fix fluentd postgresql plugin

### LINKED ISSUE #4877: Some job retry attempts missing on AKS
image

### PR DESCRIPTION
**Pull request by hzy46**
to fix #4877

### CODE REVIEW

**Comment by hzy46** [comment]
### Tested by sumitting a job with 1 task, 100 max retries:

In framework table:
image

In pods table:
image


### Tested by submitting a job with 50 tasks, 20 max retries:

In framework table:
image

In pods table (there is an additional task attempt in framework attempt id=1):

image

File: src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb [hunk 1]
```diff
@@ -140,8 +140,65 @@ def format(tag, time, record)
...
+      # select from framework_history, ensure there is no corresponding history object
+      selectResult = thread[:conn].exec_params('SELECT uid from framework_history where uid=$1', [uid])
+      if selectResult.cmd_tuples == 0
+          # if there is no existing records, try to insert a new one.
+          thread[:conn].exec_params("INSERT INTO framework_history (\"#{@insertedAt_col}\", \"#{@updatedAt_col}\", \"#{@uid_col}\", \"#{@frameworkName_col}\", \"#{@attemptIndex_col}\", \"#{@historyType_col}\", \"#{@snapshot_col}\") " +
+            "VALUES ($1, $2, $3, $4, $5, $6, $7)", [time, time, uid, frameworkName, attemptIndex, historyType, snapshot]
+          )
+      else
+        # if there is an existing record, ignore it.
+        log.warn "[pgjson] chunk #{hex_id}: ignored framework snapshot object, uid=#{uid}"
```
Comment (yqwang-ms) [inline src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb:166]: log.warn "[pgjson] chunk #{hex_id}: ignored framework snapshot object as it already exist, uid=#{uid}"

File: src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb [hunk 2]
```diff
@@ -140,8 +140,65 @@ def format(tag, time, record)
...
+      snapshot = record_value(record["objectSnapshot"])
+      selectResult = thread[:conn].exec_params('SELECT uid from pods where uid=$1', [uid])
+      if selectResult.cmd_tuples == 0
+          # if there is no existing records, try to insert a new one.
+          thread[:conn].exec("INSERT INTO pods (\"#{@insertedAt_col}\", \"#{@updatedAt_col}\", \"#{@uid_col}\", \"#{@frameworkName_col}\", \"#{@attemptIndex_col}\", \"#{@taskroleName_col}\", \"#{@taskroleIndex_col}\", \"#{@taskAttemptIndex_col}\", \"#{@snapshot_col}\") " + 
+            "VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)", [time, time, uid, frameworkName, attemptIndex, taskroleName, taskroleIndex, taskAttemptIndex, snapshot]
+          )
+      else
+        # if there is an existing record, ignore it.
+        log.warn "[pgjson] chunk #{hex_id}: ignored pod snapshot object, uid=#{uid}"
```
Comment (yqwang-ms) [inline src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb:193]: log.warn "[pgjson] chunk #{hex_id}: ignored pod snapshot object as it already exist, uid=#{uid}"

**Review by yqwang-ms — APPROVED** [review]

File: src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb (hunk 1 above)
Comment (hzy46) [reply src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb]: Done

File: src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb (hunk 2 above)
Comment (hzy46) [reply src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb]: Done

**Pull request merged**